### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by Chynchenko**
Unit tests. Not sure in DeleteTextTest especially

dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: DeleteTextTest.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        [Theory]
+        [InlineData(1)]
+        public async Task DeleteTextSuccessfull(int id)
+        {
+            var repository = new Mock<IRepositoryWrapper>();
+            var mockMapper = new Mock<IMapper>();
+
+            repository.Setup(x => x.TextRepository.GetFirstOrDefaultAsync(It.IsAny<Expression<Func<Text, bool>>>(), 
+                It.IsAny<Func<IQueryable<Text>,IIncludableQueryable<Text, object>>>()))
+                .ReturnsAsync(new Text { Id= id });
```
Comment (EyR1oN) [inline DeleteTextTest.cs:23]: Move the logic of creation object outside of return, you can use private methods or constants

File: DeleteTextTest.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+            repository.Setup(x => x.TextRepository.Delete(new Text()
+            { Id = id }));
```
Comment (EyR1oN) [inline DeleteTextTest.cs:26]: The same here

File: GetAllTextsTest.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+namespace Streetcode.XUnitTest.StreetcodeTest.TextTest
+{
+    public class GetAllTextsTest
+    { 
+    [Fact]
+    public async Task GetAlltextsSuccess()
+    {
+        var repository = new Mock<IRepositoryWrapper>();
+        var mockMapper = new Mock<IMapper>();
```
Comment (EyR1oN) [inline GetAllTextsTest.cs:19]: The same here

File: GetAllTextsTest.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        {
+            new TextDTO() { Id = 1 }
+        };
+
+        var testTexts = new Text() { Id = 1 };
+
+        repository.Setup(repo => repo.TextRepository.GetFirstOrDefaultAsync(It.IsAny<Expression<Func<Text, bool>>>(), It.IsAny<Func<IQueryable<Text>, IIncludableQueryable<Text, Text>>?>()))
+            .ReturnsAsync(testTexts);
+
+        repository.Setup(repo => repo.TextRepository.GetAllAsync(null, null)).ReturnsAsync(testTextsList);
```
Comment (EyR1oN) [inline GetAllTextsTest.cs:36]: Can we do with Any instead of nulls?

File: GetAllTextsTest.cs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        var repository = new Mock<IRepositoryWrapper>();
+        var mockMapper = new Mock<IMapper>();
+
+        repository.Setup(repo => repo.TextRepository.GetFirstOrDefaultAsync(It.IsAny<Expression<Func<Text, bool>>>(),
+            It.IsAny<Func<IQueryable<Text>, IIncludableQueryable<Text, Text>>?>()))
+            .ReturnsAsync((Text)null);
+
+        repository.Setup(repo => repo.TextRepository.GetAllAsync(null, null)).ReturnsAsync((List<Text>)null);
+
+        mockMapper.Setup(x => x.Map<IEnumerable<TextDTO>>(It.IsAny<IEnumerable<object>>())).Returns(new List<TextDTO>() { new TextDTO() { Id = 1 } });
```
Comment (EyR1oN) [inline GetAllTextsTest.cs:60]: the same here

File: GetAllTextsTest.cs [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+        Assert.NotNull(result);
+        Assert.NotEmpty(result.Value);
+    }
+
+    [Fact]
+    public async Task GetAllTextsError()
+    {
+        var repository = new Mock<IRepositoryWrapper>();
+        var mockMapper = new Mock<IMapper>();
```
Comment (EyR1oN) [inline GetAllTextsTest.cs:52]: the same here

File: GetAllTextsTest.cs [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using System.Linq.Expressions;
+using Xunit;
+
+namespace Streetcode.XUnitTest.StreetcodeTest.TextTest
+{
+    public class GetAllTextsTest
+    { 
+    [Fact]
+    public async Task GetAlltextsSuccess()
```
Comment (EyR1oN) [inline GetAllTextsTest.cs:16]: Rename methods as here https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices

File: GetTextByIdTest.cs [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+            var testText = new Text() { Id = id };
+
+            repository.Setup(repo => repo.TextRepository.GetFirstOrDefaultAsync(It.IsAny<Expression<Func<Text, bool>>>(),
+                It.IsAny<Func<IQueryable<Text>, IIncludableQueryable<Text, Text>>?>()))
+            .ReturnsAsync(testText);
+
+            mockMapper.Setup(x => x.Map<TextDTO>(It.IsAny<Text>())).Returns((Text sourceText) =>
+                {
+                    return new TextDTO { Id = sourceText.Id };
+                });
```
Comment (EyR1oN) [inline GetTextByIdTest.cs:31]: labda doesnt work as expected

File: GetTextByStreetcodeIdTest.cs [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+            var testText = new Text() { StreetcodeId = id };
+
+            repository.Setup(repo => repo.TextRepository.GetFirstOrDefaultAsync(It.IsAny<Expression<Func<Text, bool>>>(),
+                It.IsAny<Func<IQueryable<Text>, IIncludableQueryable<Text, Text>>?>()))
+            .ReturnsAsync(testText);
+
+            mockMapper.Setup(x => x.Map<TextDTO>(It.IsAny<Text>())).Returns((Text sourceText) =>
+            {
+                return new TextDTO { StreetcodeId = sourceText.StreetcodeId };
+            });
```
Comment (EyR1oN) [inline GetTextByStreetcodeIdTest.cs:41]: lambda

**Review by EyR1oN — CHANGES_REQUESTED** [review]

File: DeleteTextTest.cs [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+            var handler = new DeleteTextHandler(repository.Object);
+
+            var result = await handler.Handle(new DeleteTextCommand(id), CancellationToken.None);
+
+            Assert.NotNull(result);
+            Assert.True(result.IsSuccess);
```
[the remark this answers is not in the document]
Comment (LanchevychMaxym) [reply DeleteTextTest.cs:33]: We can not include Null assertion to Multiple

File: GetAllTextsTest.cs [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        var repository = new Mock<IRepositoryWrapper>();
+        var mockMapper = new Mock<IMapper>();
+
+        var testTextsList = new List<Text>()
+        {
+            new Text() { Id = 1 }
```
Comment (LanchevychMaxym) [inline GetAllTextsTest.cs:23]: some numbers, let's use variabled instead

File: GetAllTextsTest.cs [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+
+        repository.Setup(repo => repo.TextRepository.GetAllAsync(null, null)).ReturnsAsync((List<Text>)null);
+
+        mockMapper.Setup(x => x.Map<IEnumerable<TextDTO>>(It.IsAny<IEnumerable<object>>())).Returns(new List<TextDTO>() { new TextDTO() { Id = 1 } });
+
+        var handler = new GetAllTextsHandler(repository.Object, mockMapper.Object);
+
+        var result = await handler.Handle(new GetAllTextsQuery(), CancellationToken.None);
+
+        Assert.NotNull(result);
```
Comment (LanchevychMaxym) [inline GetAllTextsTest.cs:66]: If we mock this test to return list, let't add assert to check if we have anything there in list, I mean just list.Any()

File: GetStreetcodeArtByStreetcodeIdTest.cs [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+
+                new StreetcodeArtDTO
+                {
+                    Index = 2,
+                    StreetcodeId = 2,
+                    Streetcode = null,
+                    ArtId = 2,
+                    Art=null
+                }
+            };
```
Comment (LanchevychMaxym) [inline GetStreetcodeArtByStreetcodeIdTest.cs:72]: Why is is duplicated? we can move to separate method

File: GetStreetcodeArtByStreetcodeIdTest.cs [hunk 14]
```diff
@@ -0,0 +1,157 @@
...
+        public GetStreetcodeArtByStreetcodeIdTest()
+        {
+            repository = new Mock<IRepositoryWrapper>();
+            mockMapper = new Mock<IMapper>();
+        }
+
+        [Theory]
+        [InlineData(1)]
+
+        public async Task GetStreetcodeArtByStreetcodeId_ReturnsCorrectStreetcodeArtByStreetcodeId(int streetcodeId)
```
Comment (LanchevychMaxym) [inline GetStreetcodeArtByStreetcodeIdTest.cs:29]: Names!

**Review by LanchevychMaxym — CHANGES_REQUESTED** [review]

File: DeleteTextTest.cs [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+namespace Streetcode.XUnitTest.StreetcodeTest.TextTest
+{
+    public class DeleteTextTest
+    {
+        [Theory]
+        [InlineData(1)]
+        public async Task DeleteTextSuccessfull(int id)
+        {
+            var repository = new Mock<IRepositoryWrapper>();
+            var mockMapper = new Mock<IMapper>();
```
[the remark this answers is not in the document]
Comment (Chynchenko) [reply DeleteTextTest.cs:19]: done

File: DeleteTextTest.cs (hunk 1 above)
Comment (Chynchenko) [reply DeleteTextTest.cs:23]: done

File: DeleteTextTest.cs (hunk 2 above)
Comment (Chynchenko) [reply DeleteTextTest.cs:26]: done

File: DeleteTextTest.cs (hunk 10 above)
[the remark this answers is not in the document]
Comment (Chynchenko) [reply DeleteTextTest.cs:33]: done

File: GetAllTextsTest.cs (hunk 3 above)
Comment (Chynchenko) [reply GetAllTextsTest.cs:19]: done

File: GetAllTextsTest.cs (hunk 7 above)
Comment (Chynchenko) [reply GetAllTextsTest.cs:16]: done

File: GetTextByIdTest.cs [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+            var handler = new GetTextByIdHandler(repository.Object, mockMapper.Object);
+
+            var result = await handler.Handle(new GetTextByIdQuery(id), CancellationToken.None);
+
+            Assert.NotNull(result);
+            Assert.Equal(id, result.Value.Id);
```
[the remark this answers is not in the document]
Comment (Chynchenko) [reply GetTextByIdTest.cs:37]: but can we include Null assertions to multiple?

File: GetStreetcodeArtByStreetcodeIdTest.cs (hunk 13 above)
Comment (Chynchenko) [reply GetStreetcodeArtByStreetcodeIdTest.cs:72]: done

File: GetStreetcodeArtByStreetcodeIdTest.cs (hunk 14 above)
Comment (Chynchenko) [reply GetStreetcodeArtByStreetcodeIdTest.cs:29]: done

**Pull request closed without merging**